src/osd/PGBackend.cc

```diff
@@ -1107,3 +1099,31 @@ void PGBackend::be_compare_scrubmaps(
     }
   }
 }
+
+void PGBackend::be_large_omap_check(const map<pg_shard_t,ScrubMap*> &maps,
+  const set<hobject_t> &master_set,
+  int& large_omap_objects,
+  ostream &warnstream)
+{
+  // Find authoratative object and check large omap object flag
+  for (auto k = master_set.begin(); k != master_set.end(); ++k) {
+    object_info_t auth_oi;
+    map<pg_shard_t, shard_info_wrapper> shard_map;
+
+    inconsistent_obj_wrapper object_error{*k};
+
+    auto auth = be_select_auth_object(*k, maps, &auth_oi, shard_map,
+                                      object_error);
+    if (auth == maps.end()) {
+      continue;
+    }
+    ScrubMap::object& auth_object = auth->second->objects[*k];
+    if(auth_object.large_omap_object_found) {
+      large_omap_objects++;
+      warnstream << "Large omap object found. Object: " << *k << " Key count: "
+                 << auth_object.large_omap_object_key_count << " Size (bytes): "
+                 << auth_object.large_omap_object_value_size << '\n';
+    }
```

Comment:
Instead of be_large_omap_check() I suggest you put just the check in be_compare_scrubmaps() after the auth object is found and it is already looping through master_set.  That way you don't have to move where the master_set is created.  And you don't need to call be_select_auth_object() twice for every object.  Also, we can add an object_error code in obj_err_t (e.g. LARGE_OMAP) and do a object_error.set_large_omap().

Reply:
Never mind adding LARGE_OMAP obj_err_t because it isn't an error, but a warning.

Reply:
That puts me back inside the acting.size() check though right? I thought we were trying to avoid that as the code won't run if there is only one shard?

Reply:
Instead of using be_select_auth_object() why not just do the warning if any replica has the issue?  The only weird case would be if there is an omap inconsistency.  It might be better in an inconsistent case if any of the copies have large omap to flag it anyway.  If you repair the inconsistency, then either the large omap will remain or it won't at that point.